File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let mp = new Map();
        for(let i=0; i< nums.length;i++)
        {
            let req = target -  nums[i];
            if(mp.has(req))
            {
                return [mp.get(req),i];
            }
            mp.set(nums[i],i);
        }

    }
}

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n = nums.size();
        int i=0,j=n-1;
        int a=-1;
        while(i<=j)
        {
            int m = (i+j)/2;
            if(nums[m]>target)
            {
                j=m-1;
            }
            else
            {
                if(nums[m]==target){
                    a= m;
                    break;
                }
                i=m+1;
            }
        }
        return a;
    }
};
